src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -321,6 +328,9 @@ private void SetInputPullModePi4(int pinNumber, PinMode mode)
             bits &= ~(3u << shift);
             bits |= (pull << shift);
             gpioReg->GPPUPPDN[(pinNumber >> 4)] = bits;
+            for (int i = 0; i < 150; i++)
```

Comment:
Just curious, is this wait documented on the Pi4 datasheet? (assuming that is already available)

Reply:
@joperezr it's not on Pi3, although it does mention to wait 150 cycles when setting so that was a guess it will be needed as well after unsetting and since we got no repro after doing the delay it's presumably correct

Reply:
@joperezr : The datasheet for the Pi4 (with the same in-depth-documentation as for the Pi3) does not exist yet. It has been announced since a long time, but it's still missing. [This page](https://www.raspberrypi.org/documentation/hardware/raspberrypi/bcm2711/README.md) and the short PDF there is all there is for now.

Reply:
I see. I just asked because Ideally I would want to avoid to have workarounds like this as much as possible specially if they aren't documented on the datasheet. Because there is no datasheet for the Pi4 yet, and because we have seen improvements in tests with it, I'm fine with adding this for now and then hopefully once the datasheet is released we can see if this assumption was correct.